### ISSUE TITLE
Examples for reading from multiple topics with the same DeserializingConsumer

### ISSUE THREAD

**Issue by dylrich** [confluentinc/confluent-kafka-python#1009]
I am trying to convert our code to use the new Avro DeserializingConsumer API and am having trouble figuring out how to consume from multiple topics at the same time. With the old API, it was straightforward: We would allow the writer schema to deserialize the message, then look at the topic name and hit different code paths based on the topic. However, this new API requires us to specify a single reader schema in the DeserializingConsumer configuration. We have potentially many different reader schemas depending on the topic - is there a way for us to just pass in a mapper from topic to deserializers like we could do on our own with the old API? This question has implications related to related to #834 I suspect.

**Comment by mhowlett** [comment]
unfortunately, i believe this is not possible. It's not possible in .net or Java (pretty sure) either. I'd like to see a "dispatching" consumer API that can do this amongst other things, but that's not in the plan in the near term. It probably isn't too hard to build a custom deserializer yourself (based on our code, but adding in this capability)?. suggest that. Alternatively, the additional overhead from using multiple consumers may not be a practical issue - depends on the scenario. number of broker requests will go up linearly with number of consumers, and this is most commonly the most important thing to consider for making things efficient. so it's not a good solution if you're fully utilizing your brokers. it would also have bandwidth implications if throughput is low (there is effectively a fixed overhead per consumer).

**Comment by dylrich** [comment]
I see - thanks for the suggestions! For now I have patched the AvroDeserializer/Serializer code internally to always defer to the writer's schema and then switch on the topic name for further parsing in application code. This isn't really ideal long term but it fits our needs for now. We have a few consumers that hit a fair number of almost-the-same-schema topics and do almost-the-same work on each of them, and would like to avoid 1) more complicated deployments of those and 2) increased stress on our brokers if we did have to switch to a one-topic-per-consumer model.
